p2p/agent.go

```diff
@@ -75,6 +75,7 @@ func NewAgent(cfg config.Network, broadcastHandler HandleBroadcast, unicastHandl
 
 // Start connects into P2P network
 func (p *Agent) Start(ctx context.Context) error {
+	ready := make(chan interface{})
```

Comment:
the declaration could move to line 94.

Reply:
I intentionally add it as the first line in this func, in case in the future we added more resources into start lifecycle, and will potentially be used before fully started